examples/src/main/kotlin/ForwardingService.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.examples
+
+import org.bitcoinj.base.*
+import org.bitcoinj.core.*
+import org.bitcoinj.kits.WalletAppKit
+import org.bitcoinj.utils.BriefLogFormatter
+import org.bitcoinj.wallet.CoinSelector
+import org.bitcoinj.wallet.SendRequest
+import org.bitcoinj.wallet.Wallet
+import org.bitcoinj.wallet.listeners.WalletCoinsReceivedEventListener
+import java.io.Closeable
+import java.io.File
+import java.util.*
+import java.util.Objects.*
+import java.util.concurrent.CompletableFuture
+
+/**
+ * ForwardingService demonstrates basic usage of bitcoinj. It creates an SPV Wallet, listens on the network
+ * and when it receives coins, simply sends them onwards to the address given on the command line.
+ */
+class ForwardingService(args: Array<String>) : Closeable {
+    private var network: BitcoinNetwork? = null
+    private var forwardingAddress: Address? = null
+
+    @Volatile
+    private var kit: WalletAppKit? = null
+
+    /**
+     * Initialize by parsing the network and forwarding address command-line arguments.
+     *
+     * @param args the arguments from [.main]
+     */
+    init {
+        if (args.size >= 2) {
+            // If network was specified, validate address against network
+            network = BitcoinNetwork.fromString(args[1]).orElseThrow()
+            forwardingAddress = AddressParser.getDefault(network).parseAddress(args[0])
+        } else {
+            // Else network not-specified, extract network from address
+            forwardingAddress = AddressParser.getDefault().parseAddress(args[0])
+            //network = forwardingAddress.network() as BitcoinNetwork
+        }
+    }
+
+    /**
+     * Start the wallet and register the coin-forwarding listener.
+     */
+    fun run() {
+        println("Network: " + network!!.id())
```

Comment:
There's a PR to the Java example to simplify this.  (There are also some additional PRs to the Java example)